File: src/Atlasd/Battlenet/Protocols/Game/ChatCommands/AdminReloadCommand.cs
using Atlasd.Daemon;
using Atlasd.Localization;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Atlasd.Battlenet.Protocols.Game.ChatCommands
{
    class AdminReloadCommand : ChatCommand
    {
        public AdminReloadCommand(List<string> arguments) : base(arguments) { }

        public override bool CanInvoke(ChatCommandContext context)
        {
            return context != null && context.GameState != null && context.GameState.ActiveAccount != null;
        }

        public override void Invoke(ChatCommandContext context)
        {
            Exception e;
            string r;
            ChatEvent.EventIds eid;

            try
            {
                Settings.Load();
                e = null;
                r = Resources.AdminReloadCommandSuccess;
                eid = ChatEvent.EventIds.EID_INFO;
            }
            catch (Exception ex)
            {
                e = ex;
                r = Resources.AdminReloadCommandFailure.Replace("{exception}", e.GetType().Name);
                eid = ChatEvent.EventIds.EID_ERROR;
            }

            foreach (var kv in context.Environment)
            {
                r = r.Replace("{" + kv.Key + "}", kv.Value);
            }

            foreach (var line in r.Split(Battlenet.Common.NewLine))
                new ChatEvent(eid, context.GameState.ChannelFlags, context.GameState.Ping, context.GameState.OnlineName, line).WriteTo(context.GameState.Client);

            if (e != null)
            {
                throw e;
            }
        }
    }
}

File: src/AtlasV/Battlenet/Protocols/Game/ChatCommands/AdminReloadCommand.vb
Imports AtlasV.Daemon
Imports AtlasV.Localization
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace AtlasV.Battlenet.Protocols.Game.ChatCommands
    Class AdminReloadCommand
        Inherits ChatCommand

        Public Sub New(ByVal varRawBuffer As Byte(), ByVal varArguments As List(Of String))
            MyBase.New(varRawBuffer, varArguments)
        End Sub

        Public Overrides Function CanInvoke(ByVal varContext As ChatCommandContext) As Boolean
            Return varContext IsNot Nothing AndAlso varContext.GameState IsNot Nothing AndAlso varContext.GameState.ActiveAccount IsNot Nothing
        End Function

        Public Overrides Sub Invoke(ByVal varContext As ChatCommandContext)
            Dim e As Exception
            Dim r As String
            Dim eid As ChatEvent.EventIds

            Try
                Settings.Load()
                e = Nothing
                r = Resources.AdminReloadCommandSuccess
                eid = ChatEvent.EventIds.EID_INFO
            Catch ex As Exception
                e = ex
                r = Resources.AdminReloadCommandFailure.Replace("{exception}", e.[GetType]().Name)
                eid = ChatEvent.EventIds.EID_ERROR
            End Try

            For Each kv In varContext.Environment
                r = r.Replace("{" & kv.Key & "}", kv.Value)
            Next

            For Each line In r.Split(Battlenet.Common.NewLine)
                Call New ChatEvent(eid, varContext.GameState.ChannelFlags, varContext.GameState.Ping, varContext.GameState.OnlineName, line).WriteTo(varContext.GameState.Client)
            Next

            If e IsNot Nothing Then
                Throw e
            End If
        End Sub
    End Class
End Namespace
